### DISCUSSION TITLE
[Bug]: Cannot use pdfjs with turbopack

### DISCUSSION THREAD

**Discussion by Matts** [mozilla/pdf.js#18804] [Q&A]
### Attach (recommended) or Link to PDF file

https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/examples/learning/helloworld.pdf

### Web browser and its version

Brave

### Operating system and its version

MacOS

### PDF.js version

4.6.82

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1. Create a nextjs turbo project
2. Follow the example installation at https://mozilla.github.io/pdf.js/examples/index.html#interactive-examples
3. You will receive a compilation error from turbopack (see image below)

it seems that turbo pre-interprets import statements and tries to resolve them. I see that there is a webpack ignore statement, possibly there has to be a similar statement for turbo

![Screenshot 2024-09-26 at 17 27 29](https://github.com/user-attachments/assets/35556513-a546-41ab-8515-281c182e7e7d)

### What is the expected behavior?

This should work

### What went wrong?

compilation error

**Comment by nicolo-ribaudo** [comment]
I recommend reporting this bug to turbo pack, since that code that PDF.js uses is perfectly valid and reasonable JavaScript.

**Comment by Matts** [comment]
https://github.com/vercel/turborepo/issues/9195

**Comment by Matts** [comment]
https://github.com/vercel/next.js/issues/70555
